WordPress/openverse
# Convert modal-related components to script setup

## Fixes



Related to #4678 by @obulat

## Description


The fix for #3582 will touch several component. This PR converts those component to `script setup` to make the further review for that fix easier.

Most of the changes are just removing the extra syntax, moving the script to the top of the file, and converting the props definition to `defineProps`.

There are three other changes to the modal:
- add an exposed `close` function so that the modal can be closed programmatically from a parent file
- for the close button inside the modal, add `stopEventPropagation` so that in the future a click on the close button does not trigger closing of the modal because it's a click outside of the modal (the parent content settings modal can get closed when the top-level licence explanation modal's close button, which is technically outside of the content settings modal, is clicked)
- it's not necessary to wrap the teleport as the `VModalContent` component's root component.

## Testing Instructions

The CI should pass.
Run the app and check that the modals work as expected (the external sources modal on an image search page, the report content modal on the single result page; on mobile screen: the content settings modal on home page and on the search page)

## Checklist


      PRs) or the media properties generator (`ov just catalog/generate-docs media-props`
      for the catalog or `ov just api/generate-docs` for the API) where applicable.

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
